Flashcards.gabriel2443/Flashcards/Menus/StudySessionMenuUI.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Flashcards.Database;
+using Flashcards.Models;
+using Spectre.Console;
+
+namespace Flashcards.Menus;
+
+internal class StudySessionMenuUI
+{
+    private StudySessionDatabase studySessionDatabase = new StudySessionDatabase();
+
+    internal void StudySessionMenu()
+    {
+        var mainMenu = new MainMenuUI();
+        bool isRunning = true;
+
+        while (isRunning)
+        {
+            var select = new SelectionPrompt<string>();
+            select.Title("  [bold]Study session MENU[/]\n");
+            select.AddChoice("Go back to main menu");
+            select.AddChoice("Start study session");
+            select.AddChoice("View study session");
+            var selectedOption = AnsiConsole.Prompt(select);
+
+            switch (selectedOption)
+            {
+                case "Start study session":
+                    StartSession();
+                    break;
+
+                case "Go back to main menu":
+                    mainMenu.MainMenu();
+                    break;
+
+                case "View study session":
+                    ViewStudySession();
+                    break;
+            }
+        }
+    }
+
+    internal void StartSession()
+    {
+        Console.Clear();
+        var random = new Random();
+        var stackDatabaseManager = new StackDatabaseManager();
+        var flashDatabaseManager = new FlashcardDatabaseManager();
+        var studySession = new StudySession();
+        int score = 0;
+        int counter = 0;
+
+        AnsiConsole.WriteLine("  Please select a stack you want to study");
+
+        var cardStacks = stackDatabaseManager.GetStacks();
+        if (cardStacks.Count() == 0)
+        {
+            AnsiConsole.WriteLine("  There are no stacks to study, please add a stack if you want to start a study session");
+            return;
+        }
+        var select = new SelectionPrompt<CardStack>();
+        select.Title("Select a stack");
+        select.AddChoices(cardStacks);
+        select.AddChoice(new CardStack { CardstackId = 0, CardstackName = "Go back to menu" });
+        select.UseConverter(stackName => stackName.CardstackName);
+        var selectedCardStack = AnsiConsole.Prompt(select);
+        if (selectedCardStack.CardstackName == "Go back to menu") return;
+
+        var getFlashcards = flashDatabaseManager.ReadFlashcardsDto(selectedCardStack);
+        AnsiConsole.WriteLine($"In the stack {selectedCardStack.CardstackName.ToUpper()} are {getFlashcards.Count()} flashcards, press 'S' to start or 0 to go back to menu");
+        var input = Console.ReadLine();
+        if (string.IsNullOrEmpty(input)) return;
+        if (input == "0") return;
+        studySession.DateStart = DateTime.Now;
+        if (input.ToLower() == "s")
+        {
+            for (int i = 0; i < getFlashcards.Count; i++)
+            {
+                var randomFlashcard = random.Next(getFlashcards.Count);
+                var flashcard = getFlashcards[randomFlashcard];
```

Comment:
I found one more thing that isn't strictly necessary but I thought I'd point it out. 

I notice you're getting a random card from the list each time. This led to a weak user experience on my end, since it happened several times that I would be shown the same card two or three times in a row (since I only had three cards in a stack). I think the randomness is a great idea, but there are other ways you could implement it. 
Instead of  `var randomFlashcard = random.Next(getFlashcards.Count` you might consider using something like:
`var shuffledFlashcards = getFlashcards.OrderBy(x => random.Next()).ToList();` outside your for loop then just iterating through your new, shuffled list. That way, each session would include every card in random order. 

Just an idea for next time you want to do something like this! Lists are very powerful.

Reply:
I missed a closing parenthesis in there but you get what I mean 😅

Reply:
I  appreciate the feedback and thanks for the idea, noted 👍